### PR TITLE
Implementing ApiApp list, create commands.

### PR DESCRIPTION
**Pull request by christav**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@christav__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ctavares). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by amarzavery** [comment]
@christav The CI is failing. IT cannot find the module 'azure-arm-apiapp'. I can see that it is mentioned in package.json and the [CI did install it](https://travis-ci.org/Azure/azure-xplat-cli/jobs/66100061#L149). What could be going wrong.

One other thing. You might want to use the latest version of the package that is 0.1.1

**Comment by christav** [comment]
@amarzavery Noticed that, working on it. Could you review the code in the meantime please?

**Comment by amarzavery** [comment]
yup

**Comment by yugangw-msft** [comment]
in case it helps, is the linux file name case sensitive? If yes, then the "apiapp.js" mentioned in the package.json might not match the real file name which is "apiApp.js", with "A".

https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/apiAppManagement/package.json#L20

**Comment by christav** [comment]
@yugangw-msft That - is an excellent possibility! I don't have a unix or mac to check this on anymore, could I ask one of you guys to just try importing the module on a mac and see if it fails? If so, that is indeed the issue and can be fixed with a quick tweak to the package.json file.

**Comment by yugangw-msft** [comment]
@christav, this is not repro'd in mac, but repro'd in the linux box of Jianghao's. After fixed the file name casing, things works now.
So please go ahead to submit a node sdk fix.

File: test/commands/arm/apiapp/lib/deployLib-tests.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+var deployLib = require('../../../../../lib/commands/arm/apiapp/lib/deployLib');
+
+var requiredEnvironment = [{
+  name: 'AZURE_APIAPP_TEST_LOCATION',
+  defaultValue: 'westus'
+}];
+
+var retry = 5;
+var testPrefix = 'apiappDeploy'
+
```
Comment (yugangw-msft) [inline test/commands/arm/apiapp/lib/deployLib-tests.js]: this retry does not appear used and the testprefix was defined early on in the same file.

**Comment by yugangw-msft** [comment]
lgtm

File: test/commands/arm/apiapp/lib/deployLib-tests.js (hunk 1 above)
Comment (christav) [inline test/commands/arm/apiapp/lib/deployLib-tests.js]: Will fix.

**Comment by christav** [comment]
@yugangw-msft Travis is green, this is good to merge. Thanks!

**Pull request merged**